miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniApp.kt

```diff
@@ -42,16 +47,37 @@ abstract class MiniApp internal constructor() {
         @JvmStatic
         fun instance(): MiniApp = instance
 
-        internal fun init(
-            miniAppDownloader: MiniAppDownloader,
-            displayer: Displayer,
-            miniAppLister: MiniAppLister
-        ) {
+        /**
+         * Initializes the SDK with the specified settings.
+         * Note that you are not required to manually call this method. The SDK is automatically
+         * initialized using the settings in AndroidManifest.xml. Only call this method if you
+         * wish to re-initialize the SDK with custom settings.
+         */
+        @Suppress("LongMethod")
+        fun init(
```

Comment:
`App ID` and `Key` only involve with `ApiClient` so we could have the feature for updating the `ApiClient` only. The `context` from the content provider should be keep unchanged.

What we're doing at first is passing the `Context` from `ContentProvider` which ensures no corruption while the client app using Miniapp SDK because the `context` is still alive as long as client app is still working.

If the client developers freely pass the `context` from any place they want, our sdk cannot control it.

Reply:
That's a good point.  I was thinking maybe I can move the `ApiClient`, `MiniAppDownloader`, and `MiniAppLister` instantiation into `RealMiniApp` and then we can avoid a lot of these issues.

I'll update this PR with some changes.